src/StackExchange.Redis/RedisDatabase.cs

```diff
@@ -3003,6 +3142,48 @@ private Message GetExpiryMessage(in RedisKey key, CommandFlags flags, DateTime?
             return Message.Create(Database, flags, RedisCommand.EXPIREAT, key, seconds);
         }
 
+        private Message GetListMultiPopMessage(RedisKey[] keys, RedisValue side, long count, CommandFlags flags)
+        {
+            var args = new RedisValue[2 + keys.Length + (count == 1 ? 0 : 2)];
+            var i = 0;
+            args[i++] = keys.Length;
+            foreach (var key in keys)
+            {
+                args[i++] = key.AsRedisValue();
+            }
+
+            args[i++] = side;
+
+            if (count != 1)
+            {
+                args[i++] = RedisLiterals.COUNT;
+                args[i++] = count;
+            }
+
+            return Message.Create(Database, flags, RedisCommand.LMPOP, args);
```

Comment:
After looking at the recent issues, I'm not sure how this behaves in cluster. Can you pop across multiple servers? If not, how does it behave? This message overload will not calculate the hash correctly for _any_ keys, so I think we're lacking a proper message type here. /cc @mgravell

Reply:
Almost certainly not, this will probably break horribly if the keys don't happen to lie on the same shard as the multiplexer hits, curiously there's another multi-key command that I used this pattern from that likely has the same problem, as you said maybe let's categorize these, I'll write the prototype here, and then after this is merged I'll import that pattern elsewhere?